wgpu-hal/src/gles/egl.rs

```diff
@@ -432,13 +433,40 @@ struct Inner {
     srgb_kind: SrgbFrameBufferKind,
 }
 
+static DISPLAYS_REFERENCE_COUNT: Lazy<Mutex<HashMap<usize, usize>>> = Lazy::new(Default::default);
+
+fn increment_display_reference_count(display: khronos_egl::Display) {
+    *DISPLAYS_REFERENCE_COUNT
+        .lock()
+        .entry(display.as_ptr() as usize)
+        .or_default() += 1;
+}
+
+fn decrement_display_reference_count(display: khronos_egl::Display) -> bool {
+    let key = &(display.as_ptr() as usize);
+    let mut guard = DISPLAYS_REFERENCE_COUNT.lock();
+    let count_ref = guard.get_mut(key).unwrap();
+
+    if *count_ref > 1 {
+        *count_ref -= 1;
+
+        false
+    } else {
+        guard.remove(key);
+
+        true
+    }
+}
+
 impl Inner {
     fn create(
         flags: wgt::InstanceFlags,
         egl: Arc<EglInstance>,
         display: khronos_egl::Display,
         force_gles_minor_version: wgt::Gles3MinorVersion,
     ) -> Result<Self, crate::InstanceError> {
+        increment_display_reference_count(display);
+
```

Comment:
The need for this reference count needs to be documented here. In particular why it's fine to call `egl.initialize` repeatedly but it's not to do so with `terminate`